### PR TITLE
[5.3] Add collection drop functionality

### PR DESCRIPTION
**Pull request by josebailo**
Create the inverse of take method. The drop method returns a new collection without the specified number of items.

### CODE REVIEW

File: src/Illuminate/Support/Collection.php [hunk 1]
```diff
@@ -1322,4 +1322,19 @@ protected function getArrayableItems($items)
 
         return (array) $items;
     }
+
+    /**
+     * Drop the first or last {$limit} items.
```
Comment (GrahamCampbell) [inline src/Illuminate/Support/Collection.php]: Maybe "Drop the first or last n items." would be fine?

Comment (josebailo) [inline src/Illuminate/Support/Collection.php]: @GrahamCampbell the comment is copied from the `take` method. Should both be changed?

If you prefer that comment I change it :smiley:

Comment (GrahamCampbell) [inline src/Illuminate/Support/Collection.php]: Meh, I'm not that bothered. Was just an observation.

**Comment by taylorotwell** [comment]
I think it's best to just use slice method.

**Comment by josebailo** [comment]
Well, I understand your point of view, but after all `drop` and `take` methods are just "shortcuts" for `slice`, and there may be people that find this functionality more clear with the `drop` nomenclature rather than `slice` (like me this morning).

**Pull request closed without merging**